### PR TITLE
Fix CMVS OSX linker error

### PR DESCRIPTION
**Pull request by mfehr**

### CODE REVIEW

**Comment by mfehr** [comment]
@dymczykm thanks for your help with this!

**Comment by dymczykm** [comment]
test this please

File: cmvs_catkin/thirdparty/graclus1.2/multilevelLib/mlkkm.c [hunk 1]
```diff
@@ -17,7 +17,7 @@
 //#include <driver.h>
 //#include "../spectralLib/driver.h"
 
-extern int spectral_initialization;
+extern int spectral_initialization __attribute__((weak_import));
```
Comment (dymczykm) [inline cmvs_catkin/thirdparty/graclus1.2/multilevelLib/mlkkm.c:20]: is this also needed?

File: cmvs_catkin/CMakeLists.txt [hunk 2]
```diff
@@ -6,29 +6,40 @@ catkin_simple(ALL_DEPS_REQUIRED)
...
+
+include_directories( thirdparty/graclus1.2/multilevelLib
+                     thirdparty/graclus1.2/metisLib)
+
+# The graclus library declares these for external variables
+# that are then defined in CMVS. On OSX undefined external
+# symbols raise a linker error by default.
+# See: http://glandium.org/blog/?p=2764
+if(APPLE)
+ set(UNDEF_SYMBOL_1 "-Wl,-U,_boundary_points")
```
Comment (dymczykm) [inline cmvs_catkin/CMakeLists.txt]: indent by two chars

File: cmvs_catkin/CMakeLists.txt [hunk 3]
```diff
@@ -6,29 +6,40 @@ catkin_simple(ALL_DEPS_REQUIRED)
...
 
 # GRACLUS
-
-add_definitions(-DNUMBITS=64)
-
-include_directories(thirdparty/graclus1.2/multilevelLib thirdparty/graclus1.2/metisLib)
-
-file(GLOB GRACLUS_LIB thirdparty/graclus1.2/metisLib/*.c)
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+	add_definitions(-DNUMBITS=64)
```
Comment (dymczykm) [inline cmvs_catkin/CMakeLists.txt]: this should be also indented by 2 chars I think

File: cmvs_catkin/CMakeLists.txt [hunk 4]
```diff
@@ -6,29 +6,40 @@ catkin_simple(ALL_DEPS_REQUIRED)
...
+endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
+
+include_directories( thirdparty/graclus1.2/multilevelLib
```
Comment (dymczykm) [inline cmvs_catkin/CMakeLists.txt]: no space after (

File: cmvs_catkin/thirdparty/graclus1.2/multilevelLib/wkkm.c [hunk 5]
```diff
@@ -14,7 +14,9 @@
 #include <metis.h>
 #include <float.h>
 
-extern int cutType, memory_saving, boundary_points;
+extern int cutType __attribute__((weak_import));
+extern int memory_saving __attribute__((weak_import));
+extern int boundary_points __attribute__((weak_import));
```
Comment (dymczykm) [inline cmvs_catkin/thirdparty/graclus1.2/multilevelLib/wkkm.c:19]: weak import needed?

File: cmvs_catkin/thirdparty/graclus1.2/multilevelLib/mlkkm.c (hunk 1 above)
Comment (mfehr) [inline cmvs_catkin/thirdparty/graclus1.2/multilevelLib/mlkkm.c:20]: Probably not, but it allows the compiler to do more error checking so it surely doesn't hurt. I'm gonna leave it here.

File: cmvs_catkin/thirdparty/graclus1.2/multilevelLib/wkkm.c (hunk 5 above)
Comment (mfehr) [inline cmvs_catkin/thirdparty/graclus1.2/multilevelLib/wkkm.c:19]: same

**Comment by dymczykm** [comment]
lgtm

**Pull request merged**